packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java

```diff
@@ -155,80 +157,53 @@ protected void onDetachedFromWindow() {
+
+                if (timeDelta < 1) {
+                    timeDelta = Long.MAX_VALUE;
                 }
 
-                final long txBytesDelta = txBytes - mLastTxBytes;
-                final long rxBytesDelta = rxBytes - mLastRxBytes;
+                final long rxBytesDelta = newTotalRxTxBytes[0] - mLastRxBytes;
+                final long txBytesDelta = newTotalRxTxBytes[1] - mLastTxBytes;
 
-                if (timeDelta > 0 && txBytesDelta >= 0 && rxBytesDelta >= 0) {
-                    mTxBytes = (long) (txBytesDelta / (timeDelta / 1000f));
-                    mRxBytes = (long) (rxBytesDelta / (timeDelta / 1000f));
-                }
-                mLastTxBytes = txBytes;
-                mLastRxBytes = rxBytes;
+                rxBytes = (long) (rxBytesDelta / (timeDelta / 1000f));
+                txBytes = (long) (txBytesDelta / (timeDelta / 1000f));
+
+                mLastRxBytes = newTotalRxTxBytes[0];
+                mLastTxBytes = newTotalRxTxBytes[1];
                 mLastUpdateTime = now;
             }
 
-            mConnectionAvailable = isConnectionAvailable();
-            final boolean enabled = mLocation != 0 && mScreenOn;
-            final boolean showUpstream =
-                    mMode == MODE_UPSTREAM_ONLY || mMode == MODE_UPSTREAM_AND_DOWNSTREAM;
-            final boolean showDownstream =
-                    mMode == MODE_DOWNSTREAM_ONLY || mMode == MODE_UPSTREAM_AND_DOWNSTREAM;
-            final boolean aboveThreshold = (showUpstream && mTxBytes > mAutoHideThreshold)
-                    || (showDownstream && mRxBytes > mAutoHideThreshold);
+            final boolean enabled = mLocation != LOCATION_DISABLED && mScreenOn;
+            final boolean aboveThreshold = isAboveThreshold(rxBytes, txBytes);
+            int submode = MODE_DOWNSTREAM_ONLY;
+
             mIsActive = enabled && mAttached && (!mAutoHide || (mConnectionAvailable && aboveThreshold));
-            int submode = MODE_UPSTREAM_AND_DOWNSTREAM;
 
             clearHandlerCallbacks();
 
             if (mIsActive) {
                 CharSequence output = "";
-                if (showUpstream && showDownstream) {
-                    if (mTxBytes > mRxBytes) {
-                        output = formatOutput(mTxBytes);
-                        submode = MODE_UPSTREAM_ONLY;
-                    } else if (mTxBytes < mRxBytes) {
-                        output = formatOutput(mRxBytes);
-                        submode = MODE_DOWNSTREAM_ONLY;
-                    } else {
-                        output = formatOutput(mRxBytes);
-                        submode = MODE_UPSTREAM_AND_DOWNSTREAM;
-                    }
-                } else if (showDownstream) {
-                    output = formatOutput(mRxBytes);
-                } else if (showUpstream) {
-                    output = formatOutput(mTxBytes);
+                if ((mMode == MODE_UPSTREAM_AND_DOWNSTREAM && txBytes > rxBytes) ||
```

Comment:
what about tx = rx case? it need to have both arrows enabled.
submode = MODE_UPSTREAM_AND_DOWNSTREAM

Reply:
That is a very rare case if ever happens. But I will add the correction.